kpi/serializers/v2/asset.py

```diff
@@ -88,6 +89,11 @@ def __init__(
         super().__init__(instance=instance, data=data, **kwargs)
 
     def create(self, validated_data):
+        ProjectHistoryLog.create_from_bulk_action(
```

Comment:
we shouldn't create the logs until after we've completed the actions. if something fails, we don't want a log saying it succeeded.

Reply:
Nevermind, I'm checking the middleware code and it should run after the action was executed.